src/Findologic/Api/SortingService.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare(strict_types=1);
+
+namespace FINDOLOGIC\FinSearch\Findologic\Api;
+
+use FINDOLOGIC\FinSearch\Utils\Utils;
+use Shopware\Core\Content\Product\Events\ProductListingCriteriaEvent;
+use Shopware\Core\Content\Product\Events\ProductListingResultEvent;
+use Shopware\Core\Content\Product\SalesChannel\Listing\ProductListingSorting;
+use Shopware\Core\Content\Product\SalesChannel\Listing\ProductListingSortingRegistry;
+use Shopware\Core\Content\Product\SalesChannel\Sorting\ProductSortingCollection;
+use Shopware\Core\Content\Product\SalesChannel\Sorting\ProductSortingEntity;
+use Shopware\Core\Framework\Uuid\Uuid;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class SortingService
+{
+    public const DEFAULT_SORT = 'score';
+
+    /** @var ProductListingSortingRegistry|null */
+    private $legacySortingRegistry;
+
+    /** @var TranslatorInterface */
+    private $translator;
+
+    public function __construct(
+        ?ProductListingSortingRegistry $legacySortingRegistry,
+        TranslatorInterface $translator
+    ) {
+        $this->legacySortingRegistry = $legacySortingRegistry;
+        $this->translator = $translator;
+    }
+
+    public function handleRequest(ProductListingCriteriaEvent $event): void
+    {
+        if (!Utils::versionLowerThan('6.3.3.0')) {
+            $this->addTopResultSorting($event);
+        }
+    }
+
+    public function handleResult(ProductListingResultEvent $event): void
+    {
+        if (Utils::versionLowerThan('6.3.3.0')) {
```

Comment:
When we have the compatibility layer classes, why do we need to check the version again?

Reply:
It is easier for me to drop this for `2.0` then, since there won't be any specific versions for the `ProductListingFeaturesSubscriber`. This change is basically just a backport of the change that I've made in https://github.com/findologic/plugin-shopware-6/pull/139.

Reply:
Alright, then let's clean this with 2.0 then 👍